crates/types/src/field_color.rs

```diff
@@ -22,64 +22,27 @@ pub enum FieldColorFunction {
     Hsv,
 }
 
-#[derive(
-    Copy,
-    Clone,
-    Debug,
-    Default,
-    Deserialize,
-    Serialize,
-    PathSerialize,
-    PathDeserialize,
-    PathIntrospect,
-)]
-pub struct FieldColor {
-    pub function: FieldColorFunction,
-    pub luminance_threshold: f32,
-    pub red_chromaticity_threshold: f32,
-    pub blue_chromaticity_threshold: f32,
-    pub green_chromaticity_threshold: f32,
-    pub green_luminance_threshold: f32,
-    pub hue_low_threshold: f32,
-    pub hue_high_threshold: f32,
-    pub saturation_low_threshold: f32,
-    pub saturation_high_threshold: f32,
+#[derive(Clone, Debug, Deserialize, Serialize, PathSerialize, PathDeserialize, PathIntrospect)]
+pub struct FieldColorParameters {
+    pub luminance: RangeInclusive<u8>,
+    pub green_luminance: RangeInclusive<u8>,
+    pub red_chromaticity: RangeInclusive<f32>,
+    pub green_chromaticity: RangeInclusive<f32>,
+    pub blue_chromaticity: RangeInclusive<f32>,
+    pub hue: RangeInclusive<u16>,
```

Comment:
Theoretically, this shouldn't be inclusive, right?

Reply:
Done, changed from `360` to `359`